### DISCUSSION TITLE
Possible to sync playlists with local files?

### DISCUSSION THREAD

**Discussion by Dadoosh** [ampache/ampache#3291] [Q&A]
Hello everybody,

this may seem a silly question but bear with me. I would like to organize my playlists with an external tool (in my case MusicBee) on my computer and then push the updates to Ampache.

Say I have a playlist called 'my-playlist' on both MusicBee **and** Ampache. Is it possible to have a file 'my-playlist.m3u' and have Ampache sync said playlist against the file at regular intervals (e.g. with a cronjob?).

If there are other mechanisms to do the job, I'd be happy about suggestions.

Thanks in advance,
Dadoosh

**Comment by lachlan-00** [comment]
yes there is. the files are in the same path?

if they're the same path in musicbee and ampache you can sync / import the playlist from the file using the api/cli.

if you want to use an import job the playlist files currently need to be sitting in the library.
I use a lot of api scripts in python to do things like this.

BUT i don't see why this can't be done as a cli command. i'll add one tomorrow to the cli
```
php ./bin/cli import:Playlist etc etc
```

we have an export command but should have an import

**Reply by Dadoosh** [reply]
Sounds brilliant, thank you!

Just to avoid confustion on my part about the terminology: Is an import really what I need? Like I said, I would like to constantly update (i.e. sync obviously) exsiting playlists (and possibly add one or two now and then). However, 'importing' to me sounds like I was creating copies of the same playlist. So would importing one file over and over again lead to something like 'my-playlist', 'my-playlist1', 'my-playlist2', ... in Ampache? If not, then an import will suit me perfectly.

> the files are in the same path?

I can make them be, I suppose. My Ampache runs on a remote server together with my Nextcloud. It's a bit of a crude workaround, but I'll have MusicBee put the .m3u files in a NC folder which NC will then sync to the server. Once there, Ill add a cronjob to adapt the file paths inside of them (search/ replace), put the playlist files in the Ampache directory and finally trigger the import process. Should do the trick (I hope o.o).

Again, thank you for your efforts! Looking forward to checking this feature out :)

Cheers,
dadoosh

**Comment by ioogithub** [comment]
was this ever implemented? I could use something like this now:

php ./bin/cli import:Playlist etc etc

can it also be done with the api?

**Reply by lachlan-00** [reply]
With the cli -i / --import in run:updateCatalog

<img width="922" height="491" alt="image" src="https://github.com/user-attachments/assets/0708ddc5-3b1f-4531-9c23-96f691b67b86" />

If the files are in the catalog they will be imported. full or relative path we try to identify a few options to ID files in the database.

If you have playlist files you can import in the web interface `https://demo.ampache.dev/playlist.php?action=show_import_playlist`
